[fx] Have replace_pattern return replaced nodes

angelayi opened the pull request:
Summary: Modified replace_pattern in the subgraph rewriter to return a list of pairs of matches along with their corresponding replacement nodes in the modified graph (`List[Tuple[Match, List[Node]]]`). This allows us to easily modify the replaced nodes, including setting the metadata.

Test Plan: CI

Differential Revision: D41737056

SherlockNoMad commented on torch/fx/subgraph_rewriter.py:
```diff
@@ -193,14 +204,16 @@ def replace_pattern_with_filters(
         ``match_filter``: A function that takes in (match: InternalMatch, original_graph: Graph, pattern_graph: Graph)
             and returns a boolean indicating whether the match satisfies the condition. See matcher_utils.py for
             definition of InternalMatch.
+        ``get_replacement_nodes``: If set, causes the function to return a list
```
do we still need this flag? 
We can return replacementPattern by defatult.

angelayi replied:
oh yes, sorry forgot to remove that comment.

angelayi replied:
ooh yes, sorry forgot to remove that comment

SherlockNoMad reviewed (approved):
per comment.

angelayi committed:
[fx] Have replace_pattern return replaced nodes (#90244)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/90244

Modified replace_pattern in the subgraph rewriter to return a list of pairs of matches along with their corresponding replacement nodes in the modified graph (`List[Tuple[Match, List[Node]]]`). This allows us to easily modify the replaced nodes, including setting the metadata.

Test Plan: CI

Reviewed By: SherlockNoMad

Differential Revision: D41737056

fbshipit-source-id: 5f3a9f9be06d9df4edd11698b379c86a9a290ed5
